src/click/_termui_impl.py

```diff
@@ -424,7 +424,7 @@ def _tempfilepager(generator, cmd, color):
     """Page through text by invoking a program on a temporary file."""
     import tempfile
 
-    filename = tempfile.mktemp()
+    filename = tempfile.mkstemp()
```

Comment:
This doesn't look correct. `mkstemp` returns a tuple, but below we use `filename` as if it was a string filename...

Reply:
Also further below, shouldn't we use `subprocess.call()` instead of `os.system()`? I don't think you can put anything nasty in the temp filename, but I think it'd also fail in other cases like `cmd` containing a space..

Reply:
Treating the return as a tuple was fixed later. Using `subprocess.call` is an open issue #1476, I need to test things like this before I merge it.